src/plugins/visualizations/public/persisted_state/persisted_state.ts

```diff
@@ -0,0 +1,254 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { EventEmitter } from 'events';
```

Comment:
nit:
looks like this is not an explicit kibana dependency in package.json even-though is used in bunch of places. 
I guess we should add it to package.json.

Reply:
@Dosant  from my point of view too, but please see: #58606 

@spalger it confuse not only me =)

Reply:
Thanks for pointing me into it. I didn't know it is automatic shimming, I though it is just transitive dependency